core/src/main/java/hudson/model/UpdateSite.java

```diff
@@ -1276,6 +1277,32 @@ public boolean isNeededDependenciesCompatibleWithInstalledVersion(PluginManager.
             return getDependenciesIncompatibleWithInstalledVersion(cache).isEmpty();
         }
 
+        /**
+         * Returns true if and only if this update addressed a currently active security vulnerability.
+         *
+         * @return true if and only if this update addressed a currently active security vulnerability.
+         */
+        @Restricted(NoExternalUse.class) // Jelly
+        public boolean fixesSecurityVulnerabilities() {
+            final PluginWrapper installed = getInstalled();
+            if (installed == null) {
+                return false;
+            }
+            boolean allWarningsStillApply = true;
+            for (Warning warning : ExtensionList.lookupSingleton(UpdateSiteWarningsMonitor.class).getActivePluginWarningsByPlugin().getOrDefault(installed, Collections.emptyList())) {
```

Comment:
NIT: due to the length / complexity of the line, it could be useful to put it into a variable. This could also help a bit the debugging.

```suggestion
            List<Warning> activePluginWarnings = ExtensionList.lookupSingleton(UpdateSiteWarningsMonitor.class).getActivePluginWarningsByPlugin().getOrDefault(installed, Collections.emptyList())
            for (Warning warning : activePluginWarnings) {
```